src/main/java/fi/helsinki/cs/tmc/core/util/Folders.java

```diff
@@ -10,13 +10,19 @@
     static Path tmp = null;
 
     public static Path tempFolder() {
-        if (tmp == null) {
-            try {
-                tmp = Files.createTempDirectory("tmc-core");
-            } catch (IOException e) {
-                tmp = Paths.get(System.getProperty("java.io.tmpdir"));
+        if (tmp != null) {
+            return tmp;
+        }
+        synchronized (Folders.class) {
+            if (tmp == null) {
```

Comment:
When this line of code is executed, we allready know that tmp == null. No need for this check.